src/util/get-github-client-config.ts

```diff
@@ -22,17 +22,52 @@ interface GitHubClientConfig extends GitHubConfig {
 	gitHubClientSecret: string;
 }
 
-export async function getGitHubApiUrl(jiraHost: string, gitHubAppId: number) {
-	const gitHubClientConfig = await getGitHubClientConfigFromAppId(gitHubAppId, jiraHost);
+export async function getGitHubApiUrl(jiraHost: string, gitHubAppId: number, logger: Logger) {
+	const gitHubClientConfig = await getGitHubClientConfigFromAppId(gitHubAppId, logger, jiraHost);
 	return await booleanFlag(BooleanFlags.GHE_SERVER, GHE_SERVER_GLOBAL, jiraHost) && gitHubClientConfig
 		? `${gitHubClientConfig.apiUrl}`
 		: GITHUB_CLOUD_API_BASEURL;
 }
 
-export const getGitHubClientConfigFromAppId = async (gitHubAppId: number | undefined, jiraHost?: string): Promise<GitHubClientConfig> => {
+/**
+ * Populates proxyBaseUrl and returns the object itself (for convenient chaining)
+ * @param config - object to setup proxy in
+ * @param logger
+ * @param gitHubBaseUrl - for cloud, either undefined or "github.com"'s base url
+ * @param jiraHost - jiraHost from context
+ */
+async function setupProxyConfig<T extends GitHubConfig>(config: T, logger: Logger, jiraHost: string, gitHubBaseUrl?: string): Promise<T> {
+	if (!await booleanFlag(BooleanFlags.USE_OUTBOUND_PROXY_SKIPLIST, false, jiraHost)) {
+		return config;
+	}
+
+	if (gitHubBaseUrl && gitHubBaseUrl != GITHUB_CLOUD_BASEURL) {
+		const skipList = await stringFlag(StringFlags.OUTBOUND_PROXY_SKIPLIST, "", jiraHost);
+		let skipOutboundProxy;
+		try {
+			skipOutboundProxy = skipList
+				.split(',')
+				.filter(hostname => !!hostname)
```

Comment:
Sorry, @gxueatlassian , could you please elaborate what do you mean by "above"? About the whole block or some particular logic/line?

Reply:
I mean why do we need to `hostname => hostname.trim()` or `hostname => new URL(hostname).host.toLowerCase()` etc? As those value comes from the feature flags, we can just always put it as exact values we like, such as `https://always-lower-case.com,https://no-space-around-comma.com`

Reply:
Yeah, true, I was just trying to be a little bit more resilient against these subtle mistakes... Also, the name of the feature flag doesn't specify what exactly is expected (it is not `outbound-proxy-baseurl-skiplist`), so I guess this logic is kinda needed, otherwise we also need to rename the FF (to remove the 2nd  and 3rd maps). If we want to get rid of the 1st map, I think it is safer to convert the value into JSON in such case, so all the strings are wrapped with quotes and it is easy to spot such mistake.

Please let me know if you find this as some issue which you'd like to have been fixed or as a nit. I find the current solution OK but don't have a strong opinion and happy to adjust it if you have one ;)

Reply:
It is definitely /nit. 

However, I think the impact is super low ( if something messup in the list, the impact is we can't connect to our internal github server for testing/pollinator purpose ). All PROD use scenario should NOT skip the the proxy.

That's why I prefer avoid the code logic at all. If we didn't put the correct value in the lauchdarkly, we will find out (as it is internal github), can always correct it. Code can be simpler. 

But anyway, definitely nit, so you can leave it alone.

Reply:
> if something messup in the list, the impact is we can't connect to our internal github server for testing/pollinator purpose

True, however debugging could take some time :)

Reply:
Right. All good mate, then leave it in :)